lib/components/Editor/CustomExtensions/Image/UnsplashImagePicker.js

```diff
@@ -0,0 +1,148 @@
+import React, { useEffect, useState, useRef } from "react";
+import isEmpty from "lodash.isempty";
+import MasonryInfiniteScroller from "react-masonry-infinite";
+
+import { searchUnsplashImages } from "apis/unsplash";
+import Input from "components/Common/Input";
+import useDebounce from "hooks/useDebounce";
+
+const UnsplashImagePicker = ({ onSubmit }) => {
+  const masonryRef = useRef(null);
+
+  const [query, setQuery] = useState("");
+  const [error, setError] = useState(false);
+  const [images, setImages] = useState([]);
+  const [pageNo, setPageNo] = useState(1);
+  const [hasMore, setHasMore] = useState(true);
+  const [loading, setLoading] = useState(false);
+
+  const debouncedQuery = useDebounce(query || "latest");
+
+  useEffect(() => {
+    fetchUnsplashPhotos(1);
+  }, [debouncedQuery]);
+
+  const fetchUnsplashPhotos = async (page) => {
+    try {
+      setLoading(true);
+      setError(false);
+
+      const response = await searchUnsplashImages(page, debouncedQuery);
+      const {
+        data: { results, total_pages },
+      } = response;
+
+      if (page === 1) {
+        setHasMore(true);
+        setImages(results);
+      } else {
+        setImages([...images, ...results]);
+      }
+
+      setPageNo(page + 1);
+
+      if (page >= total_pages) {
```

Comment:
```suggestion
      setHasMore(page < total_pages)
```

We can remove this if condition and the ```setHasMore(true)``` above. All conditions can be handled by this single statement.